cli/flox-rust-sdk/src/models/search.rs

```diff
@@ -171,10 +182,12 @@ impl Query {
                     limit,
                     ..Query::default()
                 };
-                if prefer_match_name {
-                    q.match_name = Some(package.to_string());
-                } else {
-                    q.r#match = Some(package.to_string());
+                match search_strategy {
+                    SearchStrategy::Match => q.r#match = Some(package.to_string()),
+                    SearchStrategy::MatchName => q.match_name = Some(package.to_string()),
+                    SearchStrategy::MatchNameOrRelPath => {
+                        q.match_name_or_rel_path = Some(package.to_string())
```

Comment:
I think we should probably use match name whenever there's an `@`, but we can do that in a followup